### PR TITLE
HLRC: Add ILM Retry

### PR DESCRIPTION
**Pull request by original-brownbear**
* Relates #33100

--------------------

* Unfortunately I only found an easy way to test the negative case here (retry fails). We have the same issue in the rest API tests where we also seem to only test the failure though.

### CODE REVIEW

File: client/rest-high-level/src/main/java/org/elasticsearch/client/RequestConverters.java
Comment (hub-cap) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/RequestConverters.java]: These should be in some IndexLifecycleRequestConverters class. i dont think it exists yet.

**Review by hub-cap — CHANGES_REQUESTED** [review]
I dont see any docs or request converter tests. I understand that you wouldn't need to do response tests since there is no response. Other than that, it looks good.

File: client/rest-high-level/src/main/java/org/elasticsearch/client/RequestConverters.java
Comment (hub-cap) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/RequestConverters.java]: https://github.com/elastic/elasticsearch/blob/master/client/rest-high-level/src/main/java/org/elasticsearch/client/RequestConverters.java#L966

Comment (original-brownbear) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/RequestConverters.java]: I just added this to save a `toArray()` call here, seemed cleaner to just add that API, but I don't have a strong opinion here :)

**Comment by original-brownbear** [comment]
@hub-cap added request converter test :)
What kind of docs are you looking for?

**Comment by hub-cap** [comment]
@original-brownbear have a look at the other PRs. They add a asciidoc page as well as a DocumentationIT test to back the asciidoc for the API.

**Review by talevy — APPROVED** [review]
LGTM

@hub-cap I agree with you re docs, but ILM is lacking on that front in general, and I have a task to fix that. So, I will continue with HLRC ILM docs outside of this PR

**Review by hub-cap — APPROVED** [review]

**Comment by hub-cap** [comment]
as long as yall do it, im fine not blocking this.

**Comment by original-brownbear** [comment]
@talevy @hub-cap there's some test failures here but I get the same with the `index-lifecycle` branch. I'll hold off on merging here until that's fixed, if you need/want this merged earlier just merge :)

**Comment by talevy** [comment]
thanks @original-brownbear, we're looking into it

**Comment by original-brownbear** [comment]
@talevy seems you guys fixed it thanks green now after merging in `index-lifecycle` => merging!

**Pull request merged**
